src/parser/parser.yy

```diff
@@ -1736,9 +1739,11 @@ index_field
 index_field_list
     : index_field {
         $$ = new IndexFieldList();
-        std::unique_ptr<meta::cpp2::IndexFieldDef> field;
-        field.reset($1);
-        $$->addField(std::move(field));
+        if ($1) {
```

Comment:
I think you could remove the if statement refer to the rule like `opt_argument_list`

Reply:
Good, fixed!